controllers/search_controller/mongodbsearch_reconcile_helper.go

```diff
@@ -264,7 +270,32 @@ func (r *MongoDBSearchReconcileHelper) ValidateSingleMongoDBSearchForSearchSourc
 		for i, search := range searchList.Items {
 			resourceNames[i] = search.Name
 		}
-		return xerrors.Errorf("Found multiple MongoDBSearch resources for search source '%s': %s", r.db.Name(), strings.Join(resourceNames, ", "))
+		return xerrors.Errorf(
+			"Found multiple MongoDBSearch resources for search source '%s': %s", r.db.Name(),
+			strings.Join(resourceNames, ", "),
+		)
+	}
+
+	return nil
+}
+
+func (r *MongoDBSearchReconcileHelper) ValidateSearchImageVersion() error {
+	version := strings.TrimSpace(r.mdbSearch.Spec.Version)
+	if version != "" {
+		if version == unsupportedSearchVersion {
+			return xerrors.Errorf(unsupportedSearchVersionErrorFmt, unsupportedSearchVersion)
+		}
+		return nil
+	}
+
+	if r.mdbSearch.Spec.StatefulSetConfiguration == nil {
+		return nil
+	}
+
+	for _, container := range r.mdbSearch.Spec.StatefulSetConfiguration.SpecWrapper.Spec.Template.Spec.Containers {
+		if strings.Contains(container.Image, unsupportedSearchVersion) {
```

Comment:
This could trigger on any sidecar containers users add. Perhaps the if check should also test whether the container name equals the `MongotContainerName` constant in `search_construction.go`?